chain-signatures/node/src/indexer_eth.rs

```
@@ -272,7 +272,7 @@ fn sign_request_from_filtered_log(
         return Err(anyhow::anyhow!("deposit is 0"));
     }
 
-    if event.key_version != 0 {
+    if event.key_version > LATEST_MPC_KEY_VERSION {
```

@ChaoticTempest @jakmeier not related to this PR, but related to guaranteed responses. What is our strategy on invalid requests? We can try processing them anyway. In this case - with latest key version.

---

it would be a bit complex to handle invalid requests with older key versions, because then we'd have to keep a mapping of key versions. Even though we're increasing the key version for derivation path, this isn't entirely breaking anything in our MPC because we can just map key version 0 to the previous derivation path. I think we might need to do that if we're bumping the key version to 1 here.

We also need to make sure that everyone is using key version 1 in our client libraries. Or do we just keep it to 0 for now and not break the client libraries

---

Yes, since there is no breaking change for clients, they can upgrade signet.js on their own pace.

In my question, I'm talking about guaranteed responses. Now, if the key version is not supported, we are dropping the request.

---

Since we now support both 0 and 1, my question is mostly about key versions > 1. Users still can pass them.

---

Yeah, we just don't return anything. Ideally the contract should match against key versions to see if they're supported or not. If they're not, just throw an error there so the system does not need to index them. It would be a silent error if the contract doesn't do anything